leshan-core/src/main/java/org/eclipse/leshan/Link.java

```diff
@@ -60,6 +61,15 @@ public Link(String url, Map<String, ?> attributes) {
         }
     }
 
+    /**
+     * Creates a new link with a set of attributes that is attached to it.
+     * @param url the link URL
+     * @param attributes the attribute set
+     */
+    public Link(String url, AttributeSet attributes) {
+        this(url, attributes == null ? null : attributes.getMap());
```

Comment:
But don't you think this will be a very convenient constructor, since we're trying to establish a uniform way to transport and store a set of related attributes in AttributeSet?

Reply:
Maybe I'm wrong but I feel that create an artificial dependency between this 2 classes is not so good.